src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -3,20 +3,31 @@ package mesosphere.marathon.state
 import com.wix.accord.dsl._
 import com.wix.accord._
 import mesosphere.marathon.api.v2.Validation._
-import org.apache.mesos.{ Protos => Mesos }
+
+import org.apache.mesos.{ Protos => mesos }
+
 import scala.collection.immutable.Seq
 
-// TODO: trait Container and specializations?
-// Current implementation with type defaulting to DOCKER and docker to NONE makes no sense
-case class Container(
-    `type`: Mesos.ContainerInfo.Type = Mesos.ContainerInfo.Type.DOCKER,
-    volumes: Seq[Volume] = Nil,
-    docker: Option[Container.Docker] = None) {
+case class Container(`type`: mesos.ContainerInfo.Type,
```

Comment:
Yes, I tried this first. But with a tendency towards the least commotion in other files, I could not use this approach. I googled and tinkered a lot to get Formats working with this and failed. Play documentation is mute about this natural pattern. :-(

The type field is still needed for backwards compatibility. If it is not present, ContainerFormats won't work similar to the way they used to.

All that said: we could write a custom reader and writer to make it as you want. But it's more code and not the prevailing pattern in Formats. If you want this, I can go for it.